Gallery-Server/src/main/java/edu/yu/cs/gallery/GalleryResource.java

```diff
@@ -22,34 +22,41 @@
 import org.springframework.http.*;
 import org.springframework.web.reactive.function.client.WebClient;
 
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
 @Path("/galleries")
 @ApplicationScoped
 @Produces(MediaType.APPLICATION_JSON_VALUE)
 @Consumes(MediaType.APPLICATION_JSON_VALUE)
 public class GalleryResource {
-    Gallery gallery;
-    static Map<Long, URL> allServers;
-    
-    @Inject
+    @Inject 
     GalleryRepository gr;
     @Inject 
     ServerStartUp ssu;
+    @Inject 
+    Utility utility;
+    
+    @ConfigProperty(name = "serverURL")
+    String serverURL;
+
+    @ConfigProperty(name = "hubURL")
+    String hubURL;
 
     public void init(@Observes StartupEvent se) {
-        this.gallery = gr.findAll().firstResult();
+        utility.gallery = gr.findAll().firstResult();
     }
 
     @GET
-    public List<Gallery> getOnServer() {
-        return gr.listAll();
-        //return gr.find().firstResult();
+    public Gallery getOnServer() {
+        return gr.findAll().firstResult();
```

Comment:
couldnt you just return `utility.gallery`?

Reply:
Done